test/test_torch.py

```diff
@@ -7430,6 +7435,17 @@ def test_roll(self):
             self.assertEqual(expected, data.roll(1), "roll with no dims should flatten and roll.")
             self.assertEqual(expected, data.roll(1, dims=None), "roll with no dims should flatten and roll.")
 
+            # test roll over multiple dimensions
+            expected = torch.tensor([[7, 8, 5, 6], [3, 4, 1, 2]], device=device)
+            double_rolled = data.roll(shifts=(2, -1), dims=(1, 0))
+            self.assertEqual(double_rolled, expected,
+                             "should be able to roll over two dimensions, got {}".format(double_rolled))
+
+            # shifts/dims should align
+            self.assertRaisesRegex(RuntimeError, "align", lambda: data.roll(shifts=(1, 2), dims=(1)))
+            self.assertRaisesRegex(RuntimeError, "align", lambda: data.roll(shifts=(1), dims=(1, 2)))
+            self.assertRaisesRegex(RuntimeError, "align", lambda: data.roll(shifts=(), dims=1))
```

Comment:
test empty dims and both empty shifts and empty dims?

Reply:
added the tests

Reply:
FYI the new tests are empty shifts and dims. Just dims being empty but having a shift isn't an error because that causes alternate specific behavior (which is tested under the section `# test roll with no dimension specified` from previous review)